packages/components/src/ControlGroup/useControlGroup.js

```diff
@@ -21,35 +24,46 @@ export function useControlGroup(props) {
 
 	const classes = cx(styles.itemFocus, isGrid && styles.itemGrid, className);
 
-	const clonedChildren = validChildren.map((child, index) => {
-		const isFirst = index === 0;
-		const isLast = index + 1 === validChildren.length;
-		const isOnly = isFirst && isLast;
-		const isMiddle = !isFirst && !isLast;
-
-		const _key = child.key || index;
-
-		const contextStyles = cx(
-			isFirst ? (isVertical ? styles.firstRow : styles.first) : undefined,
-			isMiddle && styles.middle,
-			isLast ? (isVertical ? styles.lastRow : styles.last) : undefined,
-		);
-
-		const contextProps = {
-			isFirst,
-			isLast,
-			isMiddle,
-			isOnly,
-			isVertical,
-			styles: contextStyles,
-		};
-
-		return (
-			<ControlGroupContext.Provider key={_key} value={contextProps}>
-				{child}
-			</ControlGroupContext.Provider>
-		);
-	});
+	const clonedChildren =
+		validChildren &&
+		validChildren.map((child, index) => {
```

Comment:
This is one big whitespace change (disable whitespace changes to see) and just adds a conditional before mapping validChildren.